python/desc/skycatalogs/skyCatalogs.py

```diff
@@ -72,8 +132,42 @@ def _compress_via_mask(tbl, id_column, region):
     If there are no objects in the region, all return values are None
 
     '''
+    import time
     if region is not None:
-        mask = _compute_mask(region, tbl['ra'], tbl['dec'])
+        if isinstance(region, PolygonalRegion):        # special case
+            # Find bounding box
+            vertices = region.get_vertices_radec()  # list of (ra, dec)
+            ra_max = max([v[0] for v in vertices])
+            ra_min = min([v[0] for v in vertices])
+            dec_max = max([v[1] for v in vertices])
+            dec_min = min([v[1] for v in vertices])
+            bnd_box = Box(ra_min, ra_max, dec_min, dec_max)
+            # Compute mask for that box
+            mask = _compute_mask(bnd_box, tbl['ra'], tbl['dec'])
+            # Save index array (indices for 0's in mask array)
+            ixes = [i for i in range(mask.size) if mask[i] == False]
+            # Get compressed ra, dec
+            masked_ra = ma.array(tbl['ra'], mask=mask)
+            ra_compress = masked_ra.compressed()
+            if ra_compress.size == 0:
+                return None, None, None, None
+            dec_compress = ma.array(tbl['dec'], mask=mask).compressed()
+            # Compute polygon mask for the compressed array
+            #t0 = time.time()
+            poly_mask = _compute_mask(region, ra_compress, dec_compress)
+            #t_interval = time.time() - t0
+            #print('Time for trimmed poly mask: ', t_interval)
+            # For any non-zero elts in polygon mask, set corresponding
+            #        elt in full mask array to True.
+            #t0 = time.time()
+            for i, e in enumerate(poly_mask):
+                if e == True:
+                    mask[ixes[i]] = True
+            #t_interval = time.time() - t0
+            #print('Time for merge: ', t_interval)
```

Comment:
I think the following is equivalent and a bit easier to understand (and may be somewhat faster):
```
if all(mask):
    return None, None, None, None

# Compute polygonal region mask for unmasked items.
ra_compress = ma.array(tbl['ra'], mask=mask).compressed()
dec_compress = ma.array(tbl['dec'], mask=mask).compressed()
poly_mask = _compute_mask(region, ra_compress, dec_compress)

# Get indices of unmasked items.
ixes = np.where(~mask)

# Update mask for unmasked items with polygonal region mask.
mask[ixes] = mask[ixes] | poly_mask
```

Reply:
Yes, letting numpy do more of the work is a better idea. Thanks.

Reply:
I've implemented this and it seems fine.  I also added some code to the __main__ to try a shape other than a box.   It seems fine. To be sure I made a plot of objects in a diamond inscribed in a box region.
<img width="766" alt="Screen Shot 2022-11-07 at 5 10 41 PM" src="https://user-images.githubusercontent.com/6539463/200449505-3c955896-21d1-4aac-b1cf-ed76c3ddbc73.png">

Reply:
I've rebased to master and added a polygonal region to the row group test.  Ready for review.